src/mca/common/dstore/dstore_base.c

```diff
@@ -2680,6 +2770,13 @@ PMIX_EXPORT pmix_status_t pmix_common_dstor_store_modex(pmix_common_dstore_ctx_t
         PMIX_DESTRUCT(&pbkt);
         return PMIX_SUCCESS;
     }
+
+    rc = _acquire_ns_lock_by_name(ds_ctx, proc.nspace);
+    if (PMIX_SUCCESS != rc) {
```

Comment:
Looks like we might need this code for the error case:

`        bo->bytes = pbkt.base_ptr;
        bo->size = pbkt.bytes_used; // restore the incoming data
        pbkt.base_ptr = NULL;
        PMIX_DESTRUCT(&pbkt);
return PMIX_SUCCESS;`

Reply:
:+1:

Reply:
Added